curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/65672136535209761a5cf02b.md

```diff
@@ -0,0 +1,660 @@
+---
+id: 65672136535209761a5cf02b
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+To play the song anytime the user clicks on it, add an `onclick` attribute to the first button element. Inside the `onclick`, call the `playSong` function with `song.id`.
+
+Don't forget you need to interpolate with the dollar sign here.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests:

You should not modify the existing button and its content.

```js
assert.match(code, /<button\s+class=('|")playlist-song-info\1.*>\s*<span\s+class=\1playlist-song-title\1>\$\{song\.title\}<\/span>\s*<span\s+class=\1playlist-song-artist\1>\$\{song\.artist\}<\/span>\s*<span\s+class=\1playlist-song-duration\1>\$\{song\.duration\}<\/span>\s*<\/button>/)
```

YOu should add an `onclick` with the value `playSong(${song.id})` as the second attribute and value of the `button` element.

```js
assert.match(code, /<button\s+class=('|")playlist-song-info\1\s*onclick=\1playSong\(\$\{song\.id\}\)\1>\s*<span\s+class=\1playlist-song-title\1>\$\{song\.title\}<\/span>\s*<span\s+class=\1playlist-song-artist\1>\$\{song\.artist\}<\/span>\s*<span\s+class=\1playlist-song-duration\1>\$\{song\.duration\}<\/span>\s*<\/button>/)
```